ArduCopter/mode_autotune.cpp

```diff
@@ -106,42 +104,44 @@ bool Copter::ModeAutoTune::init(bool ignore_checks)
 
         case UNINITIALISED:
             // autotune has never been run
-            success = start(false);
-            if (success) {
-                // so store current gains as original gains
-                backup_gains_and_initialise();
-                // advance mode to tuning
-                mode = TUNING;
-                // send message to ground station that we've started tuning
-                update_gcs(AUTOTUNE_MESSAGE_STARTED);
-            }
+            // so store current gains as original gains
+            backup_gains_and_initialise();
+            // advance mode to tuning
+            mode = TUNING;
+            // send message to ground station that we've started tuning
+            update_gcs(AUTOTUNE_MESSAGE_STARTED);
             break;
 
         case TUNING:
             // we are restarting tuning after the user must have switched ch7/ch8 off so we restart tuning where we left off
-            success = start(false);
-            if (success) {
-                // reset gains to tuning-start gains (i.e. low I term)
-                load_intra_test_gains();
-                // write dataflash log even and send message to ground station
-                Log_Write_Event(DATA_AUTOTUNE_RESTART);
-                update_gcs(AUTOTUNE_MESSAGE_STARTED);
-            }
+            // reset gains to tuning-start gains (i.e. low I term)
+            load_intra_test_gains();
+            // write dataflash log even and send message to ground station
+            Log_Write_Event(DATA_AUTOTUNE_RESTART);
+            update_gcs(AUTOTUNE_MESSAGE_STARTED);
             break;
 
         case SUCCESS:
             // we have completed a tune and the pilot wishes to test the new gains in the current flight mode
             // so simply apply tuning gains (i.e. do not change flight mode)
             load_tuned_gains();
             Log_Write_Event(DATA_AUTOTUNE_PILOT_TESTING);
-            break;
+            return;
```

Comment:
Humm, this won't allow the two lines below to run. I think that isn't good as those should be initialized too.

Reply:
True.  Not sure what I was thinking there.